src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -163,33 +166,62 @@ def report_json(
             if not edge_connection:
                 get_logger().warning("Can not establish connection. Skip sending span.")
                 return duration
+
+    lock = threading.Lock()
```

Comment:
Did you test timing? maybe with a small amount of batches we don't need to create threads...
* How much time does it take to send 1 compressed batch?
* How much time does it take to send 2 compressed batches one by one?
* How much time does it take to send 2 compressed batches in parallel using threads?

I assume that we usually won't get too many batches, so lets test timing for 1-4 batches, maybe the threading overhead is more than the parallel requests save

Reply:
I added extra debug logs to log timming

Reply:
added debug logs to log timming info